src/certs.rs

```diff
@@ -182,7 +180,12 @@ async fn load_server_cert_and_key(
     cert_file: &Path,
     key_file: &Path,
 ) -> Result<(Vec<CertificateDer<'static>>, PrivateKeyDer<'static>)> {
-    let cert_reader = &mut BufReader::new(File::open(cert_file)?);
+    let cert_contents = tokio::fs::read(cert_file).await?;
+    let key_contents = tokio::fs::read(key_file).await?;
+
+    let cert_reader = &mut BufReader::new(&cert_contents[..]);
+    let key_reader = &mut BufReader::new(&key_contents[..]);
```

Comment:
since `rustls_pemfile`expectes a `std BufReader` we need to load the contents in memory and then create one to avoid blocking the tokio runtime with `std::fs` calls.